File: src/contests/cg/puzzle/haskell/1d-bush-fire.hs
-- 🎮 CodinGame Puzzle - 1d-bush-fire
-- https://www.codingame.com/training/easy/1d-bush-fire

count :: String -> Int
count = go 0
  where
    go acc [] = acc
    -- Drop at the fire covers it plus the next two cells — skip past all 3.
    go acc ('f' : rest) = go (acc + 1) (drop 2 rest)
    go acc (_ : rest) = go acc rest

main :: IO ()
main = do
  n <- readLn :: IO Int
  ls <- mapM (const getLine) [1 .. n]
  mapM_ (print . count) ls

File: src/contests/cg/puzzle/ruby/1d-bush-fire.rb
# 🎮 CodinGame Puzzle - 1d-bush-fire
# https://www.codingame.com/training/easy/1d-bush-fire

n = gets.to_i
n.times do
  strip = gets.chomp
  drops = 0
  j = 0
  while j < strip.length
    if strip[j] == "f"
      # Drop at j covers j, j+1, j+2 — skip past all 3.
      drops += 1
      j += 3
    else
      j += 1
    end
  end
  puts drops
end
